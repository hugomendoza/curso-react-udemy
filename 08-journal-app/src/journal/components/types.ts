export interface PropsNavBar {
  drawerWidth: number
}

export interface notesProps {
  id: number
  title: string
  body: string
}
