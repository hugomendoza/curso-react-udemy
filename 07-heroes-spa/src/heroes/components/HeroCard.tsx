import { Link } from "react-router-dom";

export interface PropsHeroes {
  id: string;
  superhero: string;
  publisher: string;
  alter_ego: string;
  first_appearance: string;
  characters: string
};

type CharactersHeroProps = Omit<PropsHeroes, "id" | "superhero" | "publisher" | "first_appearance">;

const CharactersByHero = ({alter_ego, characters}:CharactersHeroProps) => {
  return ( alter_ego === characters)
    ? <></>
    : <p>{ characters }</p>
};

export const HeroCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters,
}:PropsHeroes) => {

  const heroImageUrl = `/assets/heroes/${id}.jpg`;

  // const charactersByHero = (<p>{ characters }</p>);

  return (
    <div className="col animate__animated animate__fadeIn">
      <div className="card">
        <div className="row no-gutters">
          <div className="col-4">
            <img
              src={heroImageUrl}
              className="card-img"
              alt={superhero}
            />
          </div>
          <div className="col-8">
            <div className="card-body">
              <h5 className="card-title">{ superhero }</h5>
              <p className="card-text">{ alter_ego }</p>
              {/* {
                ( alter_ego !== characters ) && charactersByHero
              } */}
              <CharactersByHero alter_ego={alter_ego} characters={characters} />
              <p className="card-text">
                <small className="text-muted">{ first_appearance }</small>
              </p>
              <Link to={`/hero/${id}`}>
                MÃ¡s...
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

const defaultProps:PropsHeroes = {
  id: "",
  superhero: "",
  publisher: "",
  alter_ego: "",
  first_appearance:"",
  characters:"",
}

HeroCard.defaultProps = defaultProps;
