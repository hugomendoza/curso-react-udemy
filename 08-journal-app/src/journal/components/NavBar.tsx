import { useAppDispatch } from "../../hooks";
import {AppBar, Toolbar, IconButton, Grid, Typography} from "@mui/material";
import { LogoutOutlined, MenuOutlined} from "@mui/icons-material"

import { PropsNavBar } from "./types";
import { startLogout } from "../../store/auth";

export const NavBar = ({ drawerWidth = 240 }:PropsNavBar) => {

  const dispatch = useAppDispatch();

  const onLogout = () => {
    dispatch( startLogout() );
  }

  return (
    <AppBar
      position="fixed"
      sx={{
        width: {sm: `calc(100% - ${ drawerWidth}px)`},
        ml: {sm: `${ drawerWidth }px`}
      }}
    >
      <Toolbar>
        <IconButton
          color="inherit"
          edge="start"
          sx={{ mr:2, display: {sm: "none"} }}
        >
          <MenuOutlined />
        </IconButton>

        <Grid
          container
          direction="row"
          justifyContent="space-between"
          alignItems="center"
        >

          <Typography
            variant="h6"
            noWrap
            component="div"
          >
            JournalApp
          </Typography>

          <IconButton
            color="error"
            onClick={ onLogout }
          >
            <LogoutOutlined />
          </IconButton>

        </Grid>

      </Toolbar>
    </AppBar>
  )
}
