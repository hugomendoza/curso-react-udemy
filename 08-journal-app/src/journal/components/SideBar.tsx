import { Box, Divider, Drawer, List, Toolbar, Typography } from "@mui/material";

import { PropsNavBar, notesProps } from "./types";
import { useAppSelector } from "../../hooks";
import SideBarItem from "./SideBarItem";



export const SideBar = ({ drawerWidth }:PropsNavBar) => {
  
  const { displayName } = useAppSelector( state => state.auth );
  const { notes } = useAppSelector( state => state.journal );
  console.log("ðŸš€ ~ file: SideBar.tsx:13 ~ SideBar ~ notes", notes)

  return (
    <Box
      component="nav"
      sx={{ width: { sm: drawerWidth }, flexShrink: {sm: 0} }}
    >
      <Drawer
        variant="permanent" // Review docs for more options
        open // Can be open or open=true
        sx={{
          display: {xs: "block"},
          "& .MuiDrawer-paper": {boxSizing: "border-box", width: drawerWidth}
        }}
      >
        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component="div"
          >
            { displayName }
          </Typography>
        </Toolbar>

        <Divider />

        <List>
          {
            notes.map( (note:notesProps) => (
              <SideBarItem key={ note.id } {...note}/>
            ))
          }
        </List>

      </Drawer>
    </Box>
  )
}
