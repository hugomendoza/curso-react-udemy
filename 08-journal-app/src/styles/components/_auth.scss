.auth__main {
  align-items: center;
  background-color: $primary;
  display: flex;
  justify-content: center;
  height: 100vh;
  margin: 0;
  width: 100vw;
}

.auth__box-container {
  background-color: white;
  box-shadow: 0px 3px $dark-grey;
  border-radius: 2px;
  padding: 20px;
  width: 250px;
}

.auth__title {
  color: darken($color: $primary, $amount: 20);
  margin-bottom: 20px;
}

.auth__input {
  color: $dark-grey;
  border: 0px;
  border-bottom: 1px solid $light-grey;
  font-size: 16px;
  margin-bottom: 10px;
  height: 20px;
  width: 100%;
  transition: border-bottom .3s ease;

  &:focus {
    outline: none;
    border-bottom: 1px solid $primary;
  }

}

.auth__social-networks {
  align-items: center;
  display: flex;
  justify-content: center;
  flex-direction: column;
  padding: 20px 0;
  width: 100%;
}

.auth__alert-error {
  background-color: red;
  border-radius: 5ox;
  color: white;
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  padding: 5px;
}