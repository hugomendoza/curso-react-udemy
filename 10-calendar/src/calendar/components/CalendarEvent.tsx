import { FC } from "react"
import { Event } from "./../types/types"

interface EventProps {
  event: Event
}

export const CalendarEvent: FC<EventProps> = ( { event } ) => {

  const { title, user } = event

  return (
    <>
      <span>{ title }</span>
      <span> - { user.name } </span>
    </>
  )
}
