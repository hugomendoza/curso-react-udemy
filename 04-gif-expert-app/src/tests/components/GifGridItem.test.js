import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en componente <GifGridItem />', () => {
  
  const title = "Dragon Ball z";
  const url = "https://imagen.jpg";
  
  const wrapper = shallow(
    <GifGridItem
      title={ title }
      url={ url }
    /> 
  );
  
  test('debe de mostrar el componente <GifGridItem />', () => {
    
    expect( wrapper ).toMatchSnapshot();

  });

  test('debe de tener un pÃ¡rrafo con el title', () => {

    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( title );

  });

  test('debe de tener la imagen igual al url y alt de los props ', () => {
    
    const img = wrapper.find('img');
    // console.log( img.props() );

    expect( img.prop( 'src' ) ).toBe( url );
    expect( img.prop( 'alt' ) ).toBe( title );

  });

  test('debe de tener animate__fadeIn', () => {

    const div = wrapper.find('div');
    console.log(div.props());
    const className = div.prop('className');

    expect( className .includes('animate__fadeIn')).toBe( true );
    
  });
  
});
