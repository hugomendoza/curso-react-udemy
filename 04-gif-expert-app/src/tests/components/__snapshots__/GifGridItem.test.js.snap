// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGridItem /> debe de mostrar el componente <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Dragon Ball z"
    src="https://imagen.jpg"
  />
  <p>
     
    Dragon Ball z
     
  </p>
</div>
`;
